go/bracketsvalidator/brackets_validator.go

```diff
@@ -0,0 +1,39 @@
+// Brackets validator
+//
+// Given an expression string exp, write a program to examine whether the pairs
+// and the orders of “{“, “}”, “(“, “)”, “[“, “]” are correct in the given
+// expression.
+//
+// Examples:
+//
+//	“[()]{}{[()()]()}”       =>  true
+//	“[(])”                   =>  false
+package bracketsvalidator
+
+func Validate(exp string) bool {
+	openingBrackets := make([]rune, 0)
+
```

Comment:
here you can add edge cases validations, for instance empty string, string which starts with `]`, `}` or `)`